src/Client.php

```diff
@@ -0,0 +1,163 @@
+     * @return Response
+     */
+    private function buildResponse(
+        string $url,
+        array $headers,
+        string $method,
+        array $body,
+        array $query
+    ): Response {
+        // Process the data before making the request
+        if(!$method) { // if method is not set, set it to GET by default
+            $method = self::METHOD_GET;
+        } else { // else convert the method to uppercase
+            $method = strtoupper($method);
+        }
+        // if there are no headers but a body set header to json by default
+        if(!isset($headers['content-type']) && is_array($body) && count($body) > 0) {
+            $headers['content-type'] = 'application/json';
+        }
+        if(isset($headers['content-type'])) {
+            match ($headers['content-type']) { // Convert the body to the appropriate format
+                'application/json' => $body = json_encode($body),
+                'application/x-www-form-urlencoded' => $body = $this->flatten($body),
+                'multipart/form-data' => $body = $this->flatten($body),
+                'application/graphql' => $body = $body[0],
+                default => $body = $body,
+            };
+        }
+        if(!is_array($headers)) {
+            $headers = [];
+        }
+        foreach ($headers as $i => $header) { // convert headers to appropriate format
+            $headers[] = $i . ':' . $header;
+            unset($headers[$i]);
+        }
+        if($query) {  // if the request has a query string, append it to the request URI
+            $url .= '?' . http_build_query($query);
+        }
+        $resp_headers = [];
+        // Initialize the curl session
+        $ch = curl_init();
+        // Set the request URI
+        curl_setopt($ch, CURLOPT_URL, $url);
+        // Set the request headers
+        curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
+        // Set the request method
+        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, $method);
+        // Set the request body
+        curl_setopt($ch, CURLOPT_POSTFIELDS, $body);
+        // Save the response headers
+        curl_setopt($ch, CURLOPT_HEADERFUNCTION, function ($curl, $header) use (&$resp_headers) {
+            $len = strlen($header);
+            $header = explode(':', $header, 2);
+
+            if (count($header) < 2) { // ignore invalid headers
+                return $len;
+            }
+
+            $resp_headers[strtolower(trim($header[0]))] = trim($header[1]);
+            return $len;
+        });
+        curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 0);
+        curl_setopt($ch, CURLOPT_TIMEOUT, 15);
+        curl_setopt($ch, CURLOPT_FOLLOWLOCATION, true);
+        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+        $resp_body = curl_exec($ch); // Execute the curl session
+        $resp_status = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+
+        if (curl_errno($ch)) {
+            $error_msg = curl_error($ch);
+            print_r(curl_errno($ch));
```

Comment:
You can use the `utopia-php/cli` library and use the `Console::error(), Console::log()` etc methods to log errors and other messages

https://github.com/utopia-php/cli/blob/master/src/CLI/Console.php#L28